packages/fx-core/src/common/azure-hosting/azureAppServiceHosting.ts

```diff
@@ -0,0 +1,54 @@
+import { AzureHosting } from "./azureHosting";
+import { BicepContext, HostType } from "./interfaces";
+import { Inputs, ResourceTemplate, TokenProvider } from "@microsoft/teamsfx-api";
+import { BotBicep, PathInfo } from "../../plugins/resource/bot/constants";
+import { generateBicepFromFile } from "../tools";
+import path from "path";
+import { Void } from "../../plugins";
+import { azureWebSiteDeploy } from "./utils";
+
+const resourceId = "provisionOutputs.webAppOutput.value.botWebAppResourceId";
+const hostName = "provisionOutputs.webAppOutput.value.validDomain";
+const webAppEndpoint = "provisionOutputs.webAppOutput.value.botWebAppEndpoint";
+const endpointAsParam = "webAppProvision.outputs.botWebAppResourceId";
+
+export class AzureAppServiceHosting extends AzureHosting {
+  configurable = true;
+  hostType = HostType.AppService;
```

Comment:
I guess HostType is renamed to ServiceType in Yukun's PR. We can sync this code change after his PR is merged.